### PR TITLE
#38: Assign roles to users & edit users

### LINKED ISSUE #38: Assign Roles to Users
### User story
As an administrator, I am able assign roles to users.

### Narrative
As an administrator, I want to attach and detach roles to/from user, such that users with those roles gain access to whole sub-sections of the application.

### Out of scope
Manage the permissions covered by the roles. 

### Acceptance criteria

| # | Given | When | Then |
|---|-------|------|------|
|1|I am an administrator|I assign a role to a user|The system associates the role with this user|
|2|I am an administrator|I detach a role from a user|The system disassociates the role from this user|
|3|I am not an administrator|I try to assign a role to user|The system prevents me from associating the role with this user|
|4|I am not an administrator|I try to detach a role from a user|The system prevents me from disassociating the role with this user|

### UI Mockup

image

### PR DESCRIPTION
**Pull request by alexstojda**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**
Please provide enough information so that others can review your pull request:

Admins can assign roles to users & edit users.

### Test plan (required)

Testing is passing

### Closing issues
List all issues this PR resolves.
- Closes #38 

### Pre-Merge checklist 
Complete these items before requesting reviews
- [x] Static Analysis Rules Validated
- [x] All commits follow naming conventions 

### Sub-tasks Checklist
- [x] UI Implementation
- [x] UI Test
- [x] Backend Implementation
- [x] Backend Tests
- [x] Integration Tests
- [x] System Tests

### CODE REVIEW

**Review by Alex-Hayeur — APPROVED** [review]
The tests are good, and the functionality covers what we know about the users. Should be good to merge.

Just to confirm though, is the editing users is only from the list of all users, or from individual profiles as well?

**Comment by alexstojda** [comment]
> The tests are good, and the functionality covers what we know about the users. Should be good to merge.
> 
> Just to confirm though, is the editing users is only from the list of all users, or from individual profiles as well?

At the moment it is still unclear as our stakeholder needs to confirm if users can manage their own accounts.

File: app/Http/Controllers/UserController.php [hunk 1]
```diff
@@ -62,23 +67,43 @@ public function store(Request $request)
...
+     * @param User $user
+     * @return RedirectResponse
+     */
+    public function update(Request $request, User $user)
+    {
+        $request->validateWithBag('updateUser', [
+            'name' => ['nullable', 'string', 'max:255'],
+            'email' => ['nullable', 'string', 'email', 'max:255', 'unique:users'],
+            'roles' => ['array'],
+        ]);
```
Comment (WindOfRussia) [inline app/Http/Controllers/UserController.php:83]: We could extract this into a custom request later on

**Review by WindOfRussia — APPROVED** [review]
Works as intended

**Pull request merged**